tenacity/_asyncio.py

```diff
@@ -30,19 +30,31 @@
 WrappedFn = t.TypeVar("WrappedFn", bound=t.Callable[..., t.Awaitable[t.Any]])
 
 
-def asyncio_sleep(duration: float) -> t.Awaitable[None]:
+async def _portable_async_sleep(seconds: float) -> None:
+    # If trio is already imported, then importing it is cheap.
+    # If trio isn't already imported, then it's definitely not running, so we
+    # can skip further checks.
+    if "trio" in sys.modules:
+        # If trio is available, then sniffio is too
+        import trio
+        import sniffio
+
+        if sniffio.current_async_library() == "trio":
+            await trio.sleep(seconds)
+            return
```

Comment:
```suggestion
    if "sniffio" in sys.modules:
        import sniffio

        if sniffio.current_async_library() == "trio":
            import trio
            await trio.sleep(seconds)
            return
```

Sounds slightly better?

Reply:
Yeah it does sound better, though I suppose it may be a slight performance loss. If sniffio is in `sys.modules` then importing it will be instant. [`sniffio.current_async_library()`](https://github.com/python-trio/sniffio/blob/a1cc1692f5be32645f65ebeb33e6a138180359e0/sniffio/_impl.py#L25) doesn't look super expensive, but if we're sticking with the status quo of this function in its entirety being called every time we want to sleep then it may be worth having slightly uglier code in exchange for a minor speedup.

Reply:
Not sure I follow your reasoning here. In your code there are 2 imports each time whereas there's one or two in the code I suggested, so I'm not sure how my version is worse? 🤔

Reply:
Well your version expect trio to be the backend just because it's imported. Not sure if it's better but I admit I don't know sniffio nor trio enough to judge 😆